tests/isotonic_test.py

```diff
@@ -0,0 +1,69 @@
+# Copyright 2023 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Tests for Isotonic Regression."""
+
+from absl.testing import absltest
+
+import jax
+import jax.numpy as jnp
+import numpy as onp
+
+from jax.test_util import check_grads
+from jaxopt._src.isotonic import isotonic_l2_pav
+from jaxopt._src import test_util
+from sklearn import isotonic
+
+
+class IsotonicPavTest(test_util.JaxoptTestCase):
+  """Tests for PAV in JAX."""
+
+  def test_output_shape_and_dtype(self, n=10):
+    """Verifies the shapes and dtypes of output."""
+    y = jax.random.normal(jax.random.PRNGKey(0), (n,))
+    output = isotonic_l2_pav(y)
+    self.assertEqual(output.shape, y.shape)
+    self.assertEqual(output.dtype, y.dtype)
+
+  def test_compare_with_sklearn(self, n=10):
+    """Compares the output with the one of sklearn."""
+    y = jax.random.normal(jax.random.PRNGKey(0), (n,))
+    output = isotonic_l2_pav(y)
+    output_sklearn = jnp.array(isotonic.isotonic_regression(y, increasing=False))
+    self.assertAlmostEqual(((output - output_sklearn)**2).mean().max(), 0)
```

Comment:
Instead of using `mean().max()` to reduce to a scalar, you can use `self.assertArraysAllClose(output, output_sklearn)`.

Reply:
Done